tensorflow-framework/src/main/java/org/tensorflow/framework/initializers/Orthogonal.java

```diff
@@ -81,7 +82,11 @@ public Orthogonal(Ops tf, double gain, long seed) {
 
   /** {@inheritDoc} */
   @Override
-  public Operand<T> call(Operand<TInt64> dims, Class<T> type) {
+  public <T extends TType> Operand<T> call(Operand<TInt64> dims, Class<T> type) {
+    if (!TNumber.class.isAssignableFrom(type)) {
+      throw new IllegalArgumentException("Tensor type must be numeric: " + type.getSimpleName());
+    }
```

Comment:
Just double-checking if that's correct, the initial generic at the class level was bound to `TFloating` while now it is `TNumber`.

Same thing for a few other initializers below.

Reply:
Some of the classes were `TFloating`, some were `TNumber` or `TType` for `Initializer`.  If I change the overloaded method to `TFloating`, the compiler would complain about not overloading the abstract method.  I am open to suggestions. Right now I throw `IllegalArgumentException` which is not optimal.

Reply:
I have changed to the generic on the base class as @karllessard suggested. This eliminates this issue for the most part.
There is still a runtime check for `Zeros` and `Ones` to check if `TNumber` or `TBool`.

Reply:
(See my question elsewhere about adding a superclass `TNumberOrBool`.)